drivers/sensor/lis2dh/lis2dh_trigger.c

```diff
@@ -367,17 +376,20 @@ static void lis2dh_work_cb(struct k_work *work)
 int lis2dh_init_interrupt(const struct device *dev)
 {
 	struct lis2dh_data *lis2dh = dev->data;
+	const struct lis2dh_config *cfg = dev->config;
 	int status;
-#if DT_INST_PROP_HAS_IDX(0, irq_gpios, 1)
 	uint8_t raw[2];
-#endif
+
+	/*
+	 * Setup INT1 (for DRDY) if defined in DT
+	 */
 
 	/* setup data ready gpio interrupt */
-	lis2dh->gpio_int1 = device_get_binding(LIS2DH_INT1_GPIO_DEV_NAME);
+	lis2dh->gpio_int1 = device_get_binding(cfg->irq1_dev_name);
 	if (lis2dh->gpio_int1 == NULL) {
-		LOG_ERR("Cannot get pointer to %s device",
-			    LIS2DH_INT1_GPIO_DEV_NAME);
-		return -EINVAL;
+		LOG_INF("Cannot get pointer to irq1_dev_name");
```

Comment:
Well, actually, since we may have instances that defines irq in DT and other that don't, I'm no longer treating this as an error. Rather, I'm returnin ok (so LOG_INF is also used instead of LOG_ERR) and lis2dh->gpio_int1 is used in other calls as a matter to discriminate instances that are using interrupts.

Moreover cfg->irq1_dev_name is NULL, so I cannot use it.

Reply:
> Moreover cfg->irq1_dev_name is NULL, so I cannot use it.

Not always, right? If the GPIO driver is not compiled in, the name pointer could be non-null and the device could still not exist.

Reply:
If that's not a concern for you, OK.

Reply:
Yes, I understand there might be false positives. Nevertheless I think we might accept this.